dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelper.java

```diff
@@ -0,0 +1,273 @@
+                                                    new JSONObject(fieldVariable.value()))));
+                    properties.put("properties", jsonObject);
+                    esMappingAPI.putMapping(indexName, properties.toString());
+
+                    //Adds to the set the mapped already set for this field
+                    mappedFields.add((type.variable() + StringPool.PERIOD + field.variable())
+                            .toLowerCase());
+
+                } catch (Exception e) {
+                    handleInvalidCustomMappingError(indexName,
+                            type != null ? type.variable() + "." + field.variable() : "[]");
+                    String message = "Error setting custom index mapping from field variable "
+                            + fieldVariable.key();
+
+                    if (field != null) {
+                        message += ". Field: " + field.name();
+                    }
+
+                    if (type != null) {
+                        message += ". Content Type: " + type.name();
+                    }
+
+                    message += ". Custom mapping will be ignored for index: " + indexName;
+                    Logger.warn(ESMappingUtilHelper.class, message, e);
+                }
+            }
+        } catch (DotDataException e) {
+            Logger.warn(ESMappingUtilHelper.class,
+                    "Error setting custom index mapping for index " + indexName, e);
+        }
+        return mappedFields;
+    }
+
+    private static void addMappingForRemainingFields(final String indexName,
+            final Set<String> mappedFields) {
+        try {
+            final List<ContentType> contentTypes = contentTypeAPI.findAll();
+            contentTypes.forEach(contentType -> contentType.fields().forEach(
+                    field -> addMappingForFieldIfNeeded(indexName, contentType, field,
+                            mappedFields)));
+        } catch (DotDataException e) {
+            Logger.warnAndDebug(ESMappingUtilHelper.class,
+                    "It was not possible to get content types to map field types in Elasticsearch"
+                            + indexName, e);
+        }
+    }
+
+
+    private static void addMappingForFieldIfNeeded(final String indexName,
+            final ContentType contentType, final Field field, final Set<String> mappedFields) {
+        final String fieldVariableName = (contentType.variable() + StringPool.PERIOD + field.variable())
+                        .toLowerCase();
+        if (!mappedFields.contains(fieldVariableName)) {
+            String mappingForField = null;
+            if (field instanceof DateField || field instanceof DateTimeField
+                    || field instanceof TimeField) {
+                mappingForField = "\"type\":\"date\",\n";
+                mappingForField += "\"format\": \"yyyy-MM-dd't'HH:mm:ss||MMM d, yyyy h:mm:ss a||yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis\"\n";
```

Comment:
Codacy found an issue: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)